### ISSUE TITLE
[pt] Idea for rule: “e que” → “que” - 2022-11-05

### ISSUE THREAD

**Issue by marcoagpinto** [languagetool-org/languagetool#7394]
Hello @ricardojosehlima

Look at this sentence:
`Ele deve ter em atenção aquilo que transmite e que permite ao recetor agir corretamente.`

“e que permite(m) a(s)/o(s)/ao(s)/à(s)” → “que permite(m) a(s)/o(s)/ao(s)/à(s)”

“e” + “que” + VMIP3.+ + a(s)/o(s)/ao(s)/à(s)

If not valid in all situations, it can still be valid in most of them.

Is it a good idea to try?

If we can't make it work with all cases, at least we can write: “in some contexts you can use blah blah”.

I believe this will be a medium impact rule, and I hope it will trigger many sentences in my thesis.

😄

**Comment by ricardojosehlima** [comment]
"e que" is a construction not well regarded in standard pt-br and we eliminate the que as in "Ele disse que vai sair e que está feliz" to "Ele disse que vai sair e está feliz"

**Comment by marcoagpinto** [comment]
ahhhhhhhhhh

I will create two rules then, one for -BR and the other for -PT.

I will code them this afternoon.

Thanks!

❤️ ❤️ ❤️ ❤️ ❤️

**Comment by marcoagpinto** [comment]
“Em português (europeu) poderá remover a conjunção 'e'.”
“Em português (Brasil) poderá remover a conjunção 'que'.”
?

**Comment by ricardojosehlima** [comment]
Yes!

**Comment by marcoagpinto** [comment]
❤️

**Comment by marcoagpinto** [comment]
My first attempt for pt-BR:

```
Portuguese (Portugal): 222 total matches
Portuguese (Portugal): 582235 total sentences considered
Portuguese (Portugal): ø0.00 rule matches per sentence
```

[0.txt](https://github.com/languagetool-org/languagetool/files/9944066/0.txt)

Tons of hits, and I am brain-dead 😛 

The removal of “que” makes it a bit hard for me to see whether some results are valid or not.

Are there any POSes I should remove in the results?

This is the current rule:
```
    <!-- E QUE ESTÁ FELIZ e está feliz -->
    <rule id='E_QUE_VERBO_REMOVER_QUE_PT_BR' name="Simplificar: E que → e" type="style">
      <!-- Created by Marco A.G.Pinto with Ricardo Joseh Lima suggestions, Portuguese rule 2022-11-05 (25-JUL-2022+) -->
      <!--
      Ele disse que vai sair e que está feliz. → Ele disse que vai sair e está feliz.
      -->
      <pattern>
        <marker>
          <token>e</token>
          <token>que</token>
        </marker>
        <token postag='VMIP3.+' postag_regexp='yes'/>
      </pattern>
      <message>Em português (Brasil) poderá omitir a conjunção 'que'.</message>
      <suggestion>\1</suggestion>
      <example correction="e">Ele disse que vai sair <marker>e que</marker> está feliz.</example>
    </rule>
```

**Comment by ricardojosehlima** [comment]
For pt-br it seems fine. Except at the beginning of sentences when it's a question "E que isso quer dizer?"

**Comment by marcoagpinto** [comment]
Sorry… I lay down and fell asleep… lol 😛 

What? In just one try, it was correct?

Let me remove the starting of sentences.

**Comment by marcoagpinto** [comment]
@ricardojosehlima 

I have just fixed it:

```
Portuguese (Portugal): 215 total matches
Portuguese (Portugal): 582235 total sentences considered
Portuguese (Portugal): ø0.00 rule matches per sentence
```
[1.txt](https://github.com/languagetool-org/languagetool/files/9944524/1.txt)

It removed:
```
E que é que ela me responde?
e que tem estado guardada - buganvilias do quintal, washi tape para prender as fotos e umas letras de mad...
E que lutam.
E que prosseguem.
E que nada tem a ver com o valor da cedência em nome de algo superior, que não esmague o individuo pelo c
E que anuncia que vai poupar centenas de milhões em juros, não tem 8 milhões de euros para participar num...
E que tem você com isso?
```

Is it okay now to commit?

Thanks!

**Comment by ricardojosehlima** [comment]
Hi it's only for questions "E que é que ela me responde?" should go, but "E que nada tem a ver com o valor da cedência em nome de algo superior, que não esmague o individuo pelo c" can stay

**Comment by marcoagpinto** [comment]
ahhhhhh...

Let me try an antipattern.

**Comment by marcoagpinto** [comment]
Ahhhh...

My antipattern does nothing 😛 😛 😛 

I am going to try to fix it.

**Comment by marcoagpinto** [comment]
@ricardojosehlima 

!!!

I believe it is working:
```
Portuguese (Portugal): 220 total matches
Portuguese (Portugal): 582235 total sentences considered
Portuguese (Portugal): ø0.00 rule matches per sentence
```
[2.txt](https://github.com/languagetool-org/languagetool/files/9944612/2.txt)

Instead of removing all the entries mentioned above, it just removes the ones with question marks:

```
E que é que ela me responde?
E que tem você com isso?
```

Is it okay to commit?

Thanks!

**Comment by ricardojosehlima** [comment]
Yes!

**Comment by marcoagpinto** [comment]
Done!

https://github.com/languagetool-org/languagetool/commit/2158cc6fe3e05956815f11e20a072e8c524cf847

Brother Ricardo!

❤️ ❤️ ❤️ ❤️ ❤️ 

Tomorrow I will work on the pt-PT version.

**Comment by marcoagpinto** [comment]
@ricardojosehlima 

Ricardo!!!

Since 5am that I have been working on it (the pt-PT version).

It had tons of false positives.

This is the final result:
https://github.com/languagetool-org/languagetool/commit/925adc89daef4a5bf310acc96199f759f88f00c3

```
Portuguese (Portugal): 44 total matches
Portuguese (Portugal): 582235 total sentences considered
Portuguese (Portugal): ø0.00 rule matches per sentence
```

[_pt_pt_21.txt](https://github.com/languagetool-org/languagetool/files/9945921/_pt_pt_21.txt)


❤️ ❤️ ❤️ ❤️ ❤️ ❤️

**Comment by ricardojosehlima** [comment]
Hmm looks like many sentences of the pt-pt case are different from previous:

Esse é um objectivo muito importante e que vai ser bastante difícil de atingir.
Escrevi um livro sobre um tema completamente novo e que vai ser uma surpresa.

When I said to ommit "que" I was considering sentences that had "que" before: "Ele quer que eu saia e que eu compre um livro --> Ele quer que eu saia e compre um livro // Is Ele quer que eu saia que compre um livro ok in pt-pt??)

These sentences above don't have a previous "que" and then... it's ok in pt-br to ommit the "e" as it is in pt-pt. So these cases could be for pt-br too.

**Comment by marcoagpinto** [comment]
@ricardojosehlima 

Dear brother,

I had to completely rewrite the pt_PT rule since I had to think in a different way for "que".

In pt-PT, it sounds well replacing:
```
Esse é um objectivo muito importante e que vai ser bastante difícil de atingir.
Esse é um objectivo muito importante que vai ser bastante difícil de atingir.
```

```
Escrevi um livro sobre um tema completamente novo e que vai ser uma surpresa.
Escrevi um livro sobre um tema completamente novo que vai ser uma surpresa.
```

`Ele quer que eu saia e compre um livro.`
doesn't suggest anything because it doesn't have "e que"
😛 

When I was going to commit the rule I found out that there was already a rule dedicated to it for BR + PT:
`SIMPLIFICAR_E_QUE_QUE`

I didn't remove it since it doesn't get all the hits these two new ones do, and also, Susana worked on this one and it had new antipatterns and suggestions.

So, if the new rules fails somehow, we still have `SIMPLIFICAR_E_QUE_QUE` as a countermeasure.

❤️ ❤️ ❤️ ❤️ ❤️ ❤️

**Comment by ricardojosehlima** [comment]
Ah ok, the test sentence was
"Ele quer que eu saia e que eu compre um livro"
how do you simplify it in pt-pt? 
(a) "Ele quer que eu saia e eu compre um livro"
(b) "Ele quer que eu saia que eu compre um livro"

**Comment by marcoagpinto** [comment]
(a)

But "eu" isn't recognised by the rule, so it won't suggest anything.

Here it what the pt-PT rule checks for:

```
      <pattern>
        <marker>
          <token>e
            <exception scope='previous' postag_regexp='no' postag='SENT_START'/>
          </token>
          <token>que</token>
        </marker>
        <token postag='VMIP3.+' postag_regexp='yes'/>
        <token postag='DA.+|SPS00:DA.+|VMN0000|VMP00.+|NC.+|AQ.+' postag_regexp='yes'>
          <exception postag_regexp='no' postag='RG'/>
        </token>
      </pattern>
```

"eu" would need to be one of these to flag it: 
`<token postag='DA.+|SPS00:DA.+|VMN0000|VMP00.+|NC.+|AQ.+' postag_regexp='yes'>`

**Comment by marcoagpinto** [comment]
This is why I started working on the pt-PT at 5am and took some 10+ hours or so to have it working.

The logic for pt_PT was completely different.

**Comment by susanaboatto** [comment]
Don't we already have a similar rule to this? `E_QUE_VERBO_E_VERBO`

**Comment by marcoagpinto** [comment]
@susanaboatto 

Yes, but it doesn't detect all the hits, and I only found out when I had the -br + -pt rules written.

😄 😄 😄 😄

**Comment by marcoagpinto** [comment]
Basically, there are hits that the old rule detects that this doesn't, and the opposite.

I didn't remove the old rule because you had enhanced it months ago, so I didn't want to throw away all the work and also because of some non-detections.
